### PR TITLE
Add some comments to WindowsSceneUnderstandingObserver

### PR DESCRIPTION
**Pull request by MaxWang-MS**
## Overview
Add comments to WindowsSceneUnderstandingObserver as part of SU feature work. Also cleaned up a little.

### CODE REVIEW

File: Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs [hunk 1]
```diff
@@ -133,7 +131,7 @@ public override void Enable()
         {
             isRemoting = UnityEngine.XR.WSA.HolographicRemoting.ConnectionState == UnityEngine.XR.WSA.HolographicStreamerConnectionState.Connected;
 
-            // This will kill the background thread when we stop in editor.
+            // Kill the background thread when we stop in editor.
```
Comment (davidkline-ms) [inline Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs:134]:
```suggestion
            // Terminate the background thread when we stop in editor.
```

File: Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs [hunk 2]
```diff
@@ -133,7 +131,7 @@ public override void Enable()
...
             killToken = killTokenSource.Token;
```
Comment (davidkline-ms) [inline Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs:135]: Consider using terminateToken, etc instead

File: Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs [hunk 3]
```diff
@@ -493,30 +480,25 @@ private enum ObserverState
...
         protected virtual void SendSceneObjectAdded(SpatialAwarenessSceneObject sceneObj, Guid id)
         {
-            // Send the mesh removed event
             sceneEventData.Initialize(this, id, sceneObj);
             SpatialAwarenessSystem?.HandleEvent(sceneEventData, OnSceneObjectAdded);
         }
 
         /// <summary>
-        /// 
+        /// Setups and starts update timers
```
Comment (davidkline-ms) [inline Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs:488]:
```suggestion
        /// Sets up and starts update timers
```

File: Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs [hunk 4]
```diff
@@ -632,6 +650,10 @@ private async Task RunObserverAsync(CancellationToken cancellationToken)
                                 sceneObjects.Add(saso.Guid, saso);
                                 SendSceneObjectAdded(saso, saso.Guid);
                             }
+                            else
+                            {
+                                SendSceneObjectUpdated(saso, saso.Guid);
```
Comment (davidkline-ms) [inline Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs:655]: i believe you will need to update the sceneObjects dictionary

```
sceneObjects[saso.Guid] = saso;
```

Comment (MaxWang-MS) [reply Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs:655]: Totally agreed

Comment (davidkline-ms) [reply Assets/MRTK/Providers/Experimental/WindowsSceneUnderstanding/WindowsSceneUnderstandingObserver.cs:655]: small nit: i would move that line above the send

**Pull request merged**